### PR TITLE
Add GitHub action for creating sync reference documentation to MS Dcos PR

### PR DESCRIPTION
**Pull request by hui1110**
As title.
GitHub action works based on the module created by this [PR](https://github.com/microsoft/spring-cloud-azure/pull/922).
Fix issue: https://github.com/Azure/azure-sdk-for-java/issues/28722

### CODE REVIEW

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# This workflow is used to update the configuration properties files MS docs repo: https://github.com/MicrosoftDocs/azure-dev-docs-pr/tree/main/articles/java/spring-framework/includes/spring-cloud-azure
```
Comment (chenrujun) [inline .github/workflows/UpdatePropertiesTableInMSDocs.yaml:1]: 1. `This workflow is used to` -> `This workflow will create PR to`
2. `files MS docs repo` -> `files in MS docs repo`
3. 
```
# For example, this PR is created by this workflow: https://github.com/MicrosoftDocs/azure-dev-docs-pr/pull/3035
```

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+name: Update properties table in MS Docs
```
Comment (chenrujun) [inline .github/workflows/UpdatePropertiesTableInMSDocs.yaml:3]: `Update the configuration properties files in MS docs repo`

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+      - uses: actions/checkout@v3
+
+      - name: Generate configuration properties files
+        run: |
+          mkdir spring-cloud-azure-markdown-appdendix-generator/src/main/asciidoc
+          ./mvnw clean install -P markdown-docs -pl spring-cloud-azure-markdown-appdendix-generator -DskipTests
+
+      - name: Get project name and version
+        id: branch
```
Comment (chenrujun) [inline .github/workflows/UpdatePropertiesTableInMSDocs.yaml:43]: Can we delete this line?

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            -f sdk/spring/pom.xml
+
+      - uses: actions/checkout@v3
+
+      - name: Generate configuration properties files
+        run: |
+          mkdir spring-cloud-azure-markdown-appdendix-generator/src/main/asciidoc
+          ./mvnw clean install -P markdown-docs -pl spring-cloud-azure-markdown-appdendix-generator -DskipTests
+
+      - name: Get project name and version
```
Comment (chenrujun) [inline .github/workflows/UpdatePropertiesTableInMSDocs.yaml:42]: `Get project_version and branch_name`

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+          mvn clean install -Dmaven.javadoc.skip=true -DskipTests \
+            -Dcheckstyle.skip=true \
+            -ntp \
+            -Dspotbugs.skip=true \
+            -Drevapi.skip=true -Djacoco.skip=true \
+            -Dparallel-test-playback \
+            -Pdev \
+            -f sdk/spring/pom.xml
+
+      - uses: actions/checkout@v3
```
Comment (chenrujun) [inline .github/workflows/UpdatePropertiesTableInMSDocs.yaml:35]: ` - name: Checkout microsoft/spring-cloud-azure`

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+      - name: Get project name and version
+        id: branch
+        run: |
+          project_version=$(mvn -q -Dexec.executable="echo" -Dexec.args='${project.version}' --non-recursive exec:exec)
+          echo ::set-output name=branch_name::spring-cloud-azure_${project_version%-*}
+          echo ::set-output name=project_version::${project_version%-*}
+
+      - uses: actions/checkout@v3
+        name: Checkot  repository
```
Comment (chenrujun) [inline .github/workflows/UpdatePropertiesTableInMSDocs.yaml:50]: `Checkout MicrosoftDocs/azure-dev-docs-pr`

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+          fetch-depth: '0'
+
+      - name: Commit files changes to branch
+        run: |
+          cp -f spring-cloud-azure-markdown-appdendix-generator/src/main/asciidoc/configuration-*.md azure-dev-docs-pr/articles/java/spring-framework/includes/spring-cloud-azure
+          cd azure-dev-docs-pr
+          git config --global user.email github-actions@github.com
+          git config --global user.name github-actions
+          git add ./articles/java/spring-framework/includes/spring-cloud-azure/configuration-*.md
+          git commit -m "update Spring Cloud Azure reference appendix ${{ steps.branch.outputs.project_version }} properties table changes"
```
Comment (chenrujun) [inline .github/workflows/UpdatePropertiesTableInMSDocs.yaml:65]: `Update the configuration properties for Spring Cloud Azure: ${{ steps.branch.outputs.project_version }}. This commit is created by GitHub Action: https://github.com/microsoft/spring-cloud-azure/actions`.

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+          git push --force "https://${{ secrets.UPDATE_PROPERTIES_TABLE_IN_MS_DOCS_GITHUB_USERNAME }}:${{ secrets.UPDATE_PROPERTIES_TABLE_IN_MS_DOCS_GITHUB_ACCESS_TOKEN }}@github.com/${{ secrets.UPDATE_PROPERTIES_TABLE_IN_MS_DOCS_GITHUB_USERNAME }}/azure-dev-docs-pr.git" HEAD:${{ steps.branch.outputs.branch_name }}
```
Comment (chenrujun) [inline .github/workflows/UpdatePropertiesTableInMSDocs.yaml:66]: How about deleting `force`?

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+      - name: Create Pull Request
+        uses: vsoch/pull-request-action@master
+        env:
+          PULL_REQUEST_TOKEN: ${{ secrets.UPDATE_PROPERTIES_TABLE_IN_MS_DOCS_GITHUB_ACCESS_TOKEN }}
+          PULL_REQUEST_REPOSITORY: MicrosoftDocs/azure-dev-docs-pr
+          PULL_REQUEST_TITLE: "Update Spring Cloud Azure Reference documentation ${{ steps.branch.outputs.project_version }} changes"
```
Comment (chenrujun) [inline .github/workflows/UpdatePropertiesTableInMSDocs.yaml:73]: `Update the configuration properties for Spring Cloud Azure: ${{ steps.branch.outputs.project_version }}`

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+          PULL_REQUEST_FROM_BRANCH: ${{ secrets.UPDATE_PROPERTIES_TABLE_IN_MS_DOCS_GITHUB_USERNAME }}:${{ steps.branch.outputs.branch_name }}
+          PULL_REQUEST_BRANCH: "main"
+          PULL_REQUEST_BODY: "Update referece docs for Spring Cloud Azure: ${{ steps.branch.outputs.project_version }}."
```
Comment (chenrujun) [inline .github/workflows/UpdatePropertiesTableInMSDocs.yaml:76]: `Update the configuration properties for Spring Cloud Azure: ${{ steps.branch.outputs.project_version }}.`

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+          PULL_REQUEST_DRAFT: true
+          PASS_ON_ERROR: true # The variable PASS_IF_EXISTS is invalid, temporarily use PASS_ON_ERROR instead
```
Comment (chenrujun) [inline .github/workflows/UpdatePropertiesTableInMSDocs.yaml:78]: Could you please create an issue for this?

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+            -Dparallel-test-playback \
+            -Pdev \
+            -f sdk/spring/pom.xml
+
+      - uses: actions/checkout@v3
+        name: Checkout microsoft/spring-cloud-azure
+
+      - name: Generate configuration properties files
+        run: |
+          mkdir spring-cloud-azure-markdown-appdendix-generator/src/main/asciidoc
```
Comment (chenrujun) [inline .github/workflows/UpdatePropertiesTableInMSDocs.yaml:41]: This is not necessary.

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+          ./mvnw clean install -P markdown-docs -pl spring-cloud-azure-markdown-appdendix-generator -DskipTests
+
+      - name: Get project_version and branch_name
+        id: branch
```
Comment (chenrujun) [inline .github/workflows/UpdatePropertiesTableInMSDocs.yaml:45]: get_project_version_and_branch_name

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+          if [[ -n "$(git status -s)" ]] ;then
+            git add ./articles/java/spring-framework/includes/spring-cloud-azure/configuration-*.md
+            git commit -m "Update the configuration properties for Spring Cloud Azure: ${{ steps.branch.outputs.project_version }}. This commit is created by GitHub Action: https://github.com/microsoft/spring-cloud-azure/actions"
+            git push "https://${{ secrets.UPDATE_PROPERTIES_TABLE_IN_MS_DOCS_GITHUB_USERNAME }}:${{ secrets.UPDATE_PROPERTIES_TABLE_IN_MS_DOCS_GITHUB_ACCESS_TOKEN }}@github.com/${{ secrets.UPDATE_PROPERTIES_TABLE_IN_MS_DOCS_GITHUB_USERNAME }}/azure-dev-docs-pr.git"
+          else
+            echo "No file changes, no commits."
+          fi
+
+      - name: Create Pull Request
+        if: ${{ steps.check-whether-topic-branch-exist.outputs.topic_branch_exists == '0' }}
```
Comment (chenrujun) [inline .github/workflows/UpdatePropertiesTableInMSDocs.yaml:90]: PR is only necessary when line 81 is true.

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+          git init azure-sdk-for-java
+          cd azure-sdk-for-java
+          git remote add origin https://github.com/Azure/azure-sdk-for-java.git
+          git config core.sparsecheckout true
+          echo "sdk/spring" >> .git/info/sparse-checkout
+          echo "eng" >> .git/info/sparse-checkout
+          echo "sdk/keyvault" >> .git/info/sparse-checkout
+          echo "sdk/boms" >> .git/info/sparse-checkout
+          echo "sdk/aot" >> .git/info/sparse-checkout
+          git pull --depth=1 origin main
```
Comment (chenrujun) [inline .github/workflows/UpdatePropertiesTableInMSDocs.yaml:26]: Is it possible to use this instead? https://github.com/snow-actions/sparse-checkout
You can use another PR to do this.

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml (hunk 13 above)
Comment (hui1110) [reply .github/workflows/UpdatePropertiesTableInMSDocs.yaml:45]: updated

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml (hunk 15 above)
Comment (hui1110) [reply .github/workflows/UpdatePropertiesTableInMSDocs.yaml:26]: I will investigate.

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml (hunk 12 above)
Comment (hui1110) [reply .github/workflows/UpdatePropertiesTableInMSDocs.yaml:41]: When the directory `asciidoc` cannot be found, the file will not be created

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+          if [[ ${{ steps.check-whether-topic-branch-exist.outputs.topic_branch_exists }} -eq 1 ]] ; then
+            echo "The branch ${{ steps.get_project_version_and_branch_name.outputs.branch_name }} already exists."
+            git checkout ${{ steps.get_project_version_and_branch_name.outputs.branch_name }}
+          elif [[ ${{ steps.check-whether-topic-branch-exist.outputs.topic_branch_exists }} -eq 0 ]] ; then
+            echo "The branch ${{ steps.get_project_version_and_branch_name.outputs.branch_name }} does not exist, create."
+            git checkout -b ${{ steps.get_project_version_and_branch_name.outputs.branch_name }}
+          fi
+          cp -f ../spring-cloud-azure-markdown-appdendix-generator/src/main/asciidoc/configuration-*.md articles/java/spring-framework/includes/spring-cloud-azure
+          if [[ -n "$(git status -s)" ]] ;then
+            echo '::set-output name=whether_to_create_pr::true'
```
Comment (chenrujun) [inline .github/workflows/UpdatePropertiesTableInMSDocs.yaml:83]: nit: `whether_to_create_pr` -> `added_a_commit_in_topic_branch`.

File: .github/workflows/UpdatePropertiesTableInMSDocs.yaml [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+      - name: Create Pull Request
+        if: ${{ steps.check-whether-topic-branch-exist.outputs.topic_branch_exists == '0' && steps.commit_files_changes_to_topic_branch.outputs.whether_to_create_pr == 'true' }}
+        uses: vsoch/pull-request-action@master
+        env:
+          PULL_REQUEST_TOKEN: ${{ secrets.UPDATE_PROPERTIES_TABLE_IN_MS_DOCS_GITHUB_ACCESS_TOKEN }}
+          PULL_REQUEST_REPOSITORY: MicrosoftDocs/azure-dev-docs-pr
+          PULL_REQUEST_TITLE: "Update the configuration properties for Spring Cloud Azure: ${{ steps.get_project_version_and_branch_name.outputs.project_version }}"
+          PULL_REQUEST_FROM_BRANCH: ${{ secrets.UPDATE_PROPERTIES_TABLE_IN_MS_DOCS_GITHUB_USERNAME }}:${{ steps.get_project_version_and_branch_name.outputs.branch_name }}
+          PULL_REQUEST_BRANCH: "main"
+          PULL_REQUEST_BODY: "Update the configuration properties for Spring Cloud Azure: ${{ steps.get_project_version_and_branch_name.outputs.project_version }}."
```
Comment (chenrujun) [inline .github/workflows/UpdatePropertiesTableInMSDocs.yaml:100]: Add more description. For example:
```
This PR is created by GitHub Actions. ${link_to_action_url} 
```
`${link_to_action_url}` can be got by this way: 
https://github.community/t/get-runs-url/16921/2

**Review by chenrujun — CHANGES_REQUESTED** [review]

**Review by chenrujun — APPROVED** [review]
LGTM.

**Pull request merged**